src/uds/mod.rs

```diff
@@ -34,51 +34,33 @@ pub use security_access::*;
 
 pub use auto_uds::{UdsError};
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
-pub enum UDSErrorTy {
-    Error(UdsError),
-    Unknown(u8)
-}
-
-pub struct UDSErrorWrapper(UDSErrorTy);
-
-impl From<u8> for UDSErrorWrapper {
-    fn from(value: u8) -> Self {
-        if let Ok(e) = UdsError::try_from(value) {
-            Self(UDSErrorTy::Error(e))
-        } else {
-            Self(UDSErrorTy::Unknown(value))
-        }
-    }
-}
-
-impl EcuNRC for UDSErrorWrapper {
+impl EcuNRC for ByteWrapper<UdsError> {
     fn desc(&self) -> String {
-        match self.0 {
-            UDSErrorTy::Error(e) =>  format!("{:?}", e),
-            UDSErrorTy::Unknown(b) => format!("Unknown error code 0x{:02X?}", b),
+        match self {
+            ByteWrapper::Standard(e) =>  format!("{:?}", e),
+            ByteWrapper::NonStandard(b) => format!("Unknown error code 0x{:02X?}", b),
         }
     }
 
     fn is_ecu_busy(&self) -> bool {
-        if let UDSErrorTy::Error(e) = self.0 {
-            e == UdsError::RequestCorrectlyReceivedResponsePending
+        if let ByteWrapper::Standard(e) = self {
+            *e == UdsError::RequestCorrectlyReceivedResponsePending
         } else {
             false
         }
```

Comment:
you can probably use `matches!()` here?

Reply:
Good point. Done in ce2f547efe642f9f55f8d9cd418bdfd47db9bdbb